katcp/test/test_resource_client.py

```diff
@@ -1114,16 +1132,19 @@ def test_group_wait(self):
         result = yield group.wait('wait_sensor', 0, timeout=None, quorum=1.0)
         self.assertTrue(result)
         # Check detailed results per client
-        for client in DUT.children.values():
+        # TODO PY3 check if we can get rid of the list() call
+        for client in list(DUT.children.values()):
             self.assertTrue(result[client.name])
         result = yield group.wait('wait_sensor', 1, timeout=0.1)
         self.assertFalse(result)
-        for client in DUT.children.values():
+        # TODO PY3 check if we can get rid of the list() call
+        for client in list(DUT.children.values()):
             self.assertFalse(result[client.name])
         # Test quorum functionality
         with self.assertRaises(TypeError):
             yield group.wait('wait_sensor', 1, timeout=0.1, quorum=1.1)
-        selected_client = DUT.children.keys()[0]
+        # TODO PY3 check if we can get rid of the list() call
```

Comment:
Could possibly replace by `next(DUT.children.keys())`, although the exception will be different if the dict is empty.